### DISCUSSION TITLE
Allow to reset sort using (filter) button

### DISCUSSION THREAD

**Discussion by francoism90** [filamentphp/filament#7698] [Ideas]
It would be great to have a reset button for sorting. :)

This is useful when the default sort differs, and is not being exposed as a sort column.
If one now toggles `name`, you cannot remove this, unless when editing the url parameters.

**Comment by pboivin** [comment]
Clicking on the resource in the main navigation should bring you back to the listing page with the default Table state.

**Reply by francoism90** [reply]
@pboivin This doesn't work when using a relation manager.

**Reply by pboivin** [reply]
Right, you would need a link that brings you back to the Edit page without the query parameters.

**Reply by francoism90** [reply]
@pboivin The only problem with that, if you have any filters set, it will remove them.

Is it possible to add a button with an action like: `@click="setSort(null)")`?

**Reply by pboivin** [reply]
Maybe something like this on the table:
```php
->headerActions([
    Action::make('resetTableSort')
        ->action(function (ListRecords $livewire) {
            $livewire->tableSortColumn = null;
            $livewire->tableSortDirection = null;
        }),
])
```
